### PR TITLE
feat: add line_separator option to Grid

### PR DESCRIPTION
**Pull request by An1MuS1111**
Allow configuring row delimiters in Grid instead of hardcoding "\n", enabling support for NUL-separated output and custom line endings

### CODE REVIEW

**Comment by An1MuS1111** [comment]
hi! @sylvestre can you please approve this?

File: examples/tabs.rs [hunk 1]
```diff
@@ -25,6 +25,7 @@ fn main() {
                 tab_size: 8,
             },
             width: 80,
+            line_separator: "\n",
```
Comment (sylvestre) [inline examples/tabs.rs]: I think it should be optional

Comment (An1MuS1111) [reply examples/tabs.rs]: I'm replacing `&'static str` with `Cow<'static, str>`.

File: src/lib.rs [hunk 2]
```diff
@@ -74,6 +75,29 @@ pub struct GridOptions {
 
     /// The width to fill with the grid
     pub width: usize,
+
+    /// The string to put at the end of each line/row in the grid (`"\n"` or `"\0"`)
+    pub line_separator: Cow<'static, str>,
```
Comment (sylvestre) [inline src/lib.rs]: adding a mandatory public field is a breaking change for every user (ls, eza)

File: src/lib.rs [hunk 3]
```diff
@@ -74,6 +75,29 @@ pub struct GridOptions {
...
+}
+
+impl Default for GridOptions {
+    fn default() -> Self {
+        Self {
+            direction: Direction::LeftToRight,
+            filling: Filling::Spaces(DEFAULT_SEPARATOR_SIZE),
+            width: 0,
```
Comment (sylvestre) [inline src/lib.rs]: a default width of 0 produces a degenerate grid, is that really a sane default?

File: src/lib.rs [hunk 4]
```diff
@@ -74,6 +75,29 @@ pub struct GridOptions {
...
+            line_separator: Cow::Borrowed("\n"),
+        }
+    }
+}
+
+impl GridOptions {
+    /// Sets a custom line separator.
+    #[must_use]
+    pub fn with_line_separator(mut self, separator: impl Into<Cow<'static, str>>) -> Self {
```
Comment (sylvestre) [inline src/lib.rs]: we now have three ways to set the separator (field, this, and the two on Grid) - please keep only one, thanks

File: src/lib.rs [hunk 5]
```diff
@@ -133,6 +157,18 @@ impl<T: AsRef<str>> Grid<T> {
         grid
     }
 
+    /// Sets a custom line separator for formatting the grid.
+    #[must_use]
+    pub fn with_line_separator(mut self, separator: impl Into<Cow<'static, str>>) -> Self {
```
Comment (sylvestre) [inline src/lib.rs:141]: could you please add a test for the empty grid case with a custom separator too?

File: src/lib.rs [hunk 6]
```diff
@@ -133,6 +157,18 @@ impl<T: AsRef<str>> Grid<T> {
...
+        self.options.line_separator = separator.into();
+        self
+    }
+
+    /// Sets the line separator to put at the end of each row in the grid.
+    pub fn set_line_separator(&mut self, separator: impl Into<Cow<'static, str>>) {
```
Comment (sylvestre) [inline src/lib.rs]: set_line_separator isn't used or tested anywhere, could you drop it?

File: tests/test.rs [hunk 7]
```diff
@@ -13,6 +13,7 @@ fn no_items() {
             direction: Direction::TopToBottom,
             filling: Filling::Spaces(2),
             width: 40,
+            line_separator: "\n".into(),
```
Comment (sylvestre) [inline tests/test.rs]: now that GridOptions implements Default, `..Default::default()` here would avoid touching 40 tests for nothing.

File: src/lib.rs (hunk 2 above)
Comment (An1MuS1111) [reply src/lib.rs]: moved the `line_separator` to `Grid`. now the option can be modified using `Grid::with_line_separator`

File: src/lib.rs (hunk 6 above)
Comment (An1MuS1111) [reply src/lib.rs]: done

File: src/lib.rs (hunk 5 above)
Comment (An1MuS1111) [reply src/lib.rs:141]: added a test `empty_grid_with_custom_separator`

**Comment by An1MuS1111** [comment]
hi! @sylvestre. please let me know if I need to change anything else. If all good I will finalize the PR on https://github.com/uutils/coreutils/pull/14789.

**Comment by sylvestre** [comment]
please be patient! :)
